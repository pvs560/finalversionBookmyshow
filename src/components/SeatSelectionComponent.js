import { Modal,Button } from 'react-bootstrap';
import React from "react";
import PropTypes from "prop-types";

 export const  SeatSelectionComponent = (props)=>{

        return (<Modal
      {...props}
      size="lg"
      aria-labelledby="contained-modal-title-vcenter"
      centered
    >
      <Modal.Header closeButton>
        <Modal.Title id="contained-modal-title-vcenter">
          {props.headername} 
        </Modal.Title>
      </Modal.Header>
      <Modal.Body>
        <p>
        {props.popupbody}
        </p>
      </Modal.Body>
      <Modal.Footer>
      <Button id={props.movieid} onClick={props.onHide}>{props.footer}</Button>
      </Modal.Footer>
    </Modal>)
        
}

SeatSelectionComponent.propTypes = {
  headername:PropTypes.string.isRequired,
  popupbody:PropTypes.any.isRequired,
  footer:PropTypes.string.isRequired,
  movieid:PropTypes.string,
  onHide:PropTypes.func.isRequired
}