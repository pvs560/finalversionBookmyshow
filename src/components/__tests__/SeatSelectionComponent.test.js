import React from 'react';
import renderer from 'react-test-renderer';

import {SeatSelectionComponent} from '../SeatSelectionComponent';

describe('SeatSelectionComponent',()=>{
    test('sanpshot renders', ()=>{
        let props ={
            "show":false, 
            "onHide":()=>{},
            "headername":"testmovieheader",
            "movieid":"testid", 
            "popupbody":"testpopupbody", 
            "footer":"testBook"
        };
        const component = renderer.create(<SeatSelectionComponent {...props}/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});