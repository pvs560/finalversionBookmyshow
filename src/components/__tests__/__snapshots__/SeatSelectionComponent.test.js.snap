// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeatSelectionComponent sanpshot renders 1`] = `null`;
